vertx-sql-client-template/src/main/asciidoc/index.adoc

```diff
@@ -0,0 +1,325 @@
+= Client SQL templates
+:toc:
+
+SQL client templates is a small library design to facilitate the execution of SQL queries.
+
+== Usage
+
+To use client SQL templates add the following dependency to the _dependencies_ section of your build descriptor:
+
+* Maven (in your `pom.xml`):
+
+[source,xml]
+----
+<dependency>
+  <groupId>${maven.groupId}</groupId>
+  <artifactId>${maven.artifactId}</artifactId>
+  <version>${maven.version}</version>
+</dependency>
+----
+* Gradle (in your `build.gradle` file):
+
+[source,groovy]
+----
+dependencies {
+  compile '${maven.groupId}:${maven.artifactId}:${maven.version}'
+}
+----
+
+== Getting started
+
+Here is the simplest way to use an SQL template.
+
+SQL template consumes _named_ parameters and thus takes (by default) a map as parameters sources instead of a tuple.
+
+SQL template produces (by default) a `RowSet<Row>` like a client `PreparedQuery`. In fact the template is a thin
+wrapper for a `PreparedQuery`.
+
+[source,$lang]
+----
+{@link examples.TemplateExamples#queryExample}
+----
+
+When you need to perform an insert or update operation and you do not care of the result, you can use instead {@link io.vertx.sqlclient.template.SqlTemplate#forUpdate}:
+
+[source,$lang]
+----
+{@link examples.TemplateExamples#insertExample}
+----
+
+== Row mapping
+
+Templates produces {@link io.vertx.sqlclient.Row} as default result.
+
+You can provide a function:
```

Comment:
I'd put `to achieve row level mapping instead:` here with the full function and template call below